dali/ft/daftprogress.cpp

```diff
@@ -88,15 +88,22 @@ void DaftProgress::onProgress(unsigned __int64 sizeDone, unsigned __int64 totalS
         unsigned secsLeft = (unsigned)(timeLeft * cycleToNanoScale /1000000000);
         char temp[20];
         formatTime(temp, secsLeft);
-        displayProgress((unsigned)(sizeDone*100/totalSize), secsLeft, temp, 
-                        sizeDone/scale,totalSize/scale,scaleUnit, 
-                        (unsigned)(msGone ? (sizeDone-startSize)/msGone : 0),
+
+        unsigned percentDone = (unsigned)(totalSize ? (sizeDone*100/totalSize) : 100);
+
+        unsigned kbPerSecond = (sizeDone-startSize) / 1024;
```

Comment:
This needs to be unsigned __int64 otherwise it is quite possible it will wrap.  Maybe a different sizeDone variable.

Reply:
pushed amend